---
title: "Paper Review for Gene Expression by Mouse Inner Ear Hair Cells during Development"
output: github_document
---

# Goals and findings of the paper

Due to the importance of inner hair cell (vestibular and cochlear) in mouse's hearing ability, researchers are interested in studying its pathogenic mutation in gene expression that causes deafness. However, hair cells are sparce and difficult to isolate from surrounding cells, hence a separation procedure called FACS is executed. Additionally, high-throughput sequencing (HTS) was used to generate an unbiased quantative transcriptome study. Subsequently, a cell type-sepcific RNA sequence study on gene expression during mouse inner ear development is performed after purification. The hypothesis was confirmed in the sense that not only hair cells demonstrated differential gene expression relative to surrounding cells by having distinct location-wise expression throughout all 4 developmental stages, but also deafness genes are more highly expressed in hair cells than in surrounding cells.


# Datasets, experiments and relevant analysis

1. **Purification**: Hair and surrounding cells are purified using enzymatic treatment to dissociate cell, then FACs to separate for hair cells. More specifically, both types of cells are fluorescently labeled with Green Fluorescence Protein (GFP) and inner cells gave off higher signal intensity to differentiate from surrounding cells. Purity is quantified using number of reads for GFP+ and GFP- gene expression. 

2. **Study design and statistics**: This experiment used a matrix design consisted of three factors, namely, cell types (hair vs. surrounding), tissue sources (cochlea vs utricle), and 4 different developmental stages (E16, P0, P4, P7) for 20,207 genes across 16 samples. No biological replication was performed. However, statistical replication is available via combining samples of different levels in other factors and treating them duplicates. Read counts of all 20207 genes were calculated from 3SEQ transcriptome based quantification analysis pipeline. Nondirectional single-end sequencing of 3'-tagged mRNA with similar lengths are used to to construct HTS libraries. RPKM (Reads Per Kilobase of transcript per Million mapped reads) and FPKM (Fragments Per Kilobase of transcript per Million mapped reads) are used to quantify gene expression.

3. **Principal component analysis (PCA)**: PCA was used to uncover the primary determinants of gene expression. Gene expression values with total number of reads across ages greater than or equal to 10 were normalized and put into a log2 transformed matrix with a minimum value of 1 for every age. This matrix contains gene expression information for GFP+ hair cells and GFP- surrounding cells. The first component accounts for 22% of total variance is strongly associated with cell type shown by a biplot of the first two principal components. The second component explains 17% of total variance has weak association with development stages. Lastly, the third principal component has no discernable correlation with any identifiable factors which suggests a highly complex gene expression pattern for both hair and surrouding cells during development.

4. **Comparing differential mRNA expression**: mRNA expressions in hair and surrounding cells are differentiated using pairwise comparison of GFP+ and GFP- samples with normalized read counts at 4 developmental stages. Higher reads correspond to higher mRNA amount. Three groups are available based on the ratio between GFP+ and GFP- (also known the as FC- fold-change value). HC-enriched gene group has a ratio higher than 2. SC-enriched gene group has a ratio between 0.5 and 2. Nonspecific gene group has a ratio lower than 0.5. Cochlear and utricular hair cells are also separately examined across all 4 developmental stages. The FC value the relative magnitude of enrichment of each gene in hair cells, averaged across organs and ages.

5. **Quantative Polymerase Chain Reaction (qPCR) validation**: qPCR was used to examine selected hair cell gene expressions (Pcdh15, Pou4f3, Cdh23, Atoh1, and Myo7A). It validated the result presented by the HTS profile which showed a significant different expression pattern for cochlear samples from stage E16 to P7. HTS expression profile was visually presented using a histogram of normalized counts for both hair and surrounding cells. There exists a relative reduction in GFP+ cell gene expression from embryonic to postnal cochleae.

6. **False Discovery Rate (FDR) for multiple comparison**: Mutiple testing corrections in this experiment used the Benjamini and Hochberg method. FDR represents the null hypothesis' type I error rate in multiple comparisons. It is calculated for each gene. 

7. **Hierarchical Clustering**: Genes are clustered into groups based on similarity of spatial and temporal expression patterns in 16 conditions. dChip was used to order genes based on expression pattern. A hierarchical clustering heat map demonstrated subsets of genes enriched in both hair (cochlear and utricular) and surrouding cells. Transcriptor factor genes like Atoh1, Gfi1, Pou4f3, and Lhx3 are more highly expressed in hair cells. Another cluster has genes that are highly expressed in cochlear hair cells but not in utricular hair cells whereas a third cluster shows the opposite pattern. Cluster distance is determined by the correlation of the expression of the 3'-tags of any selected two genes across all samples. The distance between a gene and its cluster and that between 2 clusters are calculated using centroid linkage method with log-transformed normalized expression values obtained via DESeq analysis. More specifically, expression values are standardized to have mean 0 and variance 1, then it is averaged to create an expression vector used to compute distance between clusters.


# Comments and discussions
This study primarily focused on examining differential gene expression between hair (cochlear vs. utricular) and surrounding cells in mouse's inner ear as well as identifying highly expressed genes in hair cells that are also potentially associated with deafness. Specifically, Kir2, Grxcr2, XIRP2 are highly enriched genes in hair cells that are also associated with deafness. I mostly agree with their interpretations of experimental results. Gene function/network analyses were present in this study since they mentioned that groups of genes that are differentially expressed were linked to function. The statistical tools used in this study could be improved. For instance, an Exploratory Data Analysis (EDA) step should be taken before any detailed statistical analysis. Secondly, the usage of PCA for dimension reduction and clustering is questionable because PCA assumes that the principle components are a linear combination of the original features and that the principle components are orthogonal. If those assumptions are violated, then results are not sensible to interpret. Hence, an assumption-checking step should be conducted before putting PCA into practice in this study. Lastly, PCA reduces data dimension by projecting a given dataset onto a transformed subspace based on variables' eigenvalues and eigenvectors. Such a project inevitably decreases variables' interpretability since only the resulting principal components could be interpreted based on total variance reduction.

